fabfile/tasks/helpers.py

```diff
@@ -806,6 +806,38 @@ def setup_hugepages_node(*args):
 def setup_hugepages():
     setup_hugepages_node(env.host_string)
 
+@task
+def setup_coremask_node(*args):
+    """Setup core mask on one or list of nodes
+    USAGE: fab setup_coremask_node:user@host1,user@host2,...
+    """
+    vrouter_file = '/etc/contrail/supervisord_vrouter_files/contrail-vrouter-dpdk.ini'
+    taskset_param = ''
+
+    for host_string in args:
+        dpdk = getattr(env, 'dpdk', None)
+        if dpdk:
+            # do not crash when coremask was not specified in testbed
+            try:
+                coremask = dpdk[env.host_string]['coremask']
+            except KeyError:
+                return
+        else:
+            return
+
+        if (coremask == ''):
+            return
+
+        # supported coremask format: hex: (0x3f); list: (0,3-5), (0,1,2,3,4,5)
+        # if specified as a list of cpus, -c flag must be provided to taskset
+        if ',' in coremask or '-' in coremask:
+            taskset_param = ' -c'
+
+        # prepend taskset only if valid coremask was specified in testbed
+        with settings(host_string=host_string):
+            sudo('sed -i \'s/command=/command=taskset%s %s /\' %s' \
+                %(taskset_param, coremask, vrouter_file))
```

Comment:
>If the core mask is specifed in the same line as huge pages (per server), it is fine.

Yes, the section in testbed.py looks like this:
env.dpdk = {
    host1: { 'huge_pages': '50', 'coremask': '0,2-6' },
    host2: { 'huge_pages': '50', 'coremask': '0xf' },
}

I'm testing changes you asked for (validation and making the parameter mandatory) and will commit it soon.